### PR TITLE
switch the documentation to run with `numpy>=2`

### PR DESCRIPTION
**Pull request by keewis**
Doctests can't really support backwards compatibility with older versions of `numpy` – the `repr` of scalars was changed – so with this we basically stop supporting builds of the docs and running the doctests with `numpy<2.0`. I don't think that's an issue, though.

For `cross` we'll need to decide what to do... `numpy>=2.0` ~stopped supporting~ deprecated `np.cross` on 2D arrays, and in #8993 I simply removed the tests for 2D arrays from the tests. If we do that here, as well, we'll have to modify the docstring. Another option would be to zero-pad to 3-dimensional.

I avoided a deciding on `np.linspace(da1_0d, da2_0d, n)` (see #9043) by extracting the value using `.item()`.

~Other than that, there seems to be an issue in `to_dask_dataframe`, which I'm still investigating.~ seems like my environment was the issue? CI doesn't fail, at least.

This is waiting on #9136

- [ ] User visible changes (including notable bug fixes) are documented in `whats-new.rst`

### CODE REVIEW

**Comment by max-sixty** [comment]
Thank you v much for doing these @keewis

**Review by JessicaS11 — APPROVED** [review]
Thanks for tackling this, @keewis!

File: ci/requirements/doc.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ dependencies:
   - nbsphinx
   - netcdf4>=1.5
   - numba
-  - numpy>=1.21,<2
+  - numpy>=1.21
```
Comment (dcherian) [inline ci/requirements/doc.yml:24]:
```suggestion
  - numpy>=2
```

File: xarray/core/computation.py [hunk 2]
```diff
@@ -1587,15 +1587,6 @@ def cross(
     array([-3,  6, -3])
     Dimensions without coordinates: dim_0
 
-    Vector cross-product with 2 dimensions, returns in the perpendicular
-    direction:
```
Comment (keewis) [inline xarray/core/computation.py:1591]: might also need to change the description in the docstring, a bit further up? It still says 1, 2, or 3 dimensional vectors are supported.

**Pull request merged**